assets/component-card.css

```diff
@@ -362,3 +362,81 @@
 .card-article-info {
   margin-top: 1rem;
 }
+
+/* Card Shapes */
+
+.card--shape .card__content {
+  padding-top: 0;
+}
+
+.card--shape.card--standard:not(.card--text) .card__inner {
+  background-color: transparent;
```

Comment:
I see, so if I understand correctly, it's mostly because we use background-2 as the default for cards and because transparent pngs are annoying.

For the first, I'd argue that a user can simply change the card background from background-2 you can still create the [proper effect](https://screenshot.click/10-06-fi38d-fd5dm.mp4) for most images and that on it's own doesn't warrant the magic overrides that break other settings.

The second is trickier for sure, but I guess I'd ask if it's common for someone with transparent assets to want to give the images a background or if it's more common they're trying to achieve [this effect](https://screenshot.click/10-23-7t69i-qtirr.png). That's just my take, but I'll defer to you and Meli here because you would know better. In which case, I'll disregard and do a final tophatting.

Reply:
I actually didn't realize there were so many other settings in general theme settings related to product cards.

The issue seems to be that sometimes you want the card background to show behind the shape (e.g. if the product card has a border, colour scheme, shadow, etc.) but most of the time you don't.

I'm ok to let the options be incompatible. When would you ever want an image shape with a rounded blue padded area behind it?

Maybe there's middle ground where we leave the background color if `card_style == 'card'` since, to me, that signals they want the card background to be prominent and probably intend for it to show through the shape.

Reply:
> Maybe there's middle ground where we leave the background color if card_style == 'card' since, to me, that signals they want the card background to be prominent and probably intend for it to show through the shape.

Yep, that's what we do today: 

<img width="375" alt="Screenshot 2023-03-10 at 1 50 18 PM" src="https://user-images.githubusercontent.com/1202812/224401152-86c42be2-8f91-4213-86ae-d618538519da.png">

It doesn't work nicely with transparent images, but I don't think there's a solution for that without introducing some other color into the mix somewhere (which I think would lead to more workarounds and unpredictability).

Reply:
While I actually think you can create [good looking use cases](https://screenshot.click/10-58-s44ap-8blk4.png) with standard cards and shapes and we should allow for that, my main argument can really be summed up with _"I think the use cases we're targeting with the logic are valid, I don't feel they warrant the exceptions in the code that we will be responsible for testing and maintaining forever"_

The argument I'd make for transparent images not working well is, transparent product images are a "shape" in their own right. It seems like an edge case that someone with transparent images would then want to further mask that image with another shape. I recognize the card background makes the shape show up better, but even that result seems odd to me. _"If I have a transparent image, why is there grey behind it?"_

Reply:
@kmeleta I do agree with @kjellr that in most cases merchants using the shapes won't want the `card__inner` background to show up in a square around the shape. Selecting `background-2` to match the page background isn't ideal from a UX or CSS perspective since it should really be transparent in that case.

With that in mind, do you have any code suggestions, e.g. only apply this override if card padding is 0?